### DISCUSSION TITLE
Using `pandoc.write` with `--extract-media`

### DISCUSSION THREAD

**Discussion by CodeSandwich** [jgm/pandoc#9594] [Q&A] [answered]
I'm converting an ODT document with images to markdown. I need to put the extracted images in the `./assets` directory. To do that I'm passing the `--extract-media=assets` flag to pandoc. It works well, the images are stored in `assets/Pictures` and the markdown file has image tags pointing at the `assets` directory `![](assets/Pictures/<hash>.<ext>)`.

How can I change the media directory if I want to use `pandoc.write`? The lua filter sees the `Image` objects with `src` unaware of the `--extract-media`, they are pointing at the plain `Pictures/<hash>.<ext>`. The media bag too has all the assets with `path` pointing at `Pictures/<hash>.<ext>`. It seems that `--extract-media` is the writer property, but I can't find it in `WriterOptions` or `CommonState` (or `ReaderOptions`). Do I need to manually modify each `Image` object to point at `assets/...` and then use `pandoc.mediabag.write("assets")`? An extra question: how can I have all the images stored directly in `assets` instead of `assets/Pictures`?

**Comment by jgm** [comment]
Does this help?
https://pandoc.org/lua-filters.html#module-pandoc.mediabag

**Reply by CodeSandwich** [reply]
Thanks, but unfortunately it doesn't. I read it before writing the questions and it doesn't answer them.

**Reply by jgm** [reply, marked as answer]
The `Pictures/` is there because it's part of the relative path in the ODT container.
You could remove it from all the Image src and change the path in the mediabag entries.

**Reply by CodeSandwich** [reply]
So to replicate `--extract-media=<path>` when using `pandoc.write` I need to modify all the `Image` objects to prefix all source paths with `<path>` and then `pandoc.mediabag.write` them to `<path>`. I was hoping for an easier solution, but that will work. TBF it should be doable in just a few lines of code. Thanks for your help!
